test/test_utils.rb

```diff
@@ -166,4 +166,18 @@ class TestUtils < JekyllUnitTest
     end
   end
 
+  context "The \`Utils.append_permalink_ending\` method" do
+    should "handle built-in permalink styles" do
+      assert_equal "/:basename/", Utils.append_permalink_ending("/:basename", :pretty)
```

Comment:
you wouldn't.  The value of template is not user-supplied, it's hardcoded in the method calls in `document.rb` and `page.rb` so we always control what the value is, and the docs for the method explicitly say that template should not have a trailing slash or extension.